storage/storage-client/src/lib.rs

```diff
@@ -275,6 +277,51 @@ impl StorageRead for StorageReadServiceClient {
             .boxed();
         Ok(stream)
     }
+
+    // TODO migrate this to `ConfigStorage` trait as non-async
+    // and implement for LibraDBTrait once `StorageRead` is deprecated
+    async fn batch_fetch_config(&self, access_paths: Vec<AccessPath>) -> Vec<Option<Vec<u8>>> {
+        let mut requests = vec![];
+        // some access paths can have the same address, so don't duplicate requests for them
+
+        let mut req_addresses = vec![];
+        for path in access_paths.iter() {
+            let address = path.address;
+            if !req_addresses.contains(&address) {
```

Comment:
nit: might be ok for now(we have few subscriptions), but do we want to avoid O(N*N) implementation here

Reply:
I originally had a HashSet tracking down seen addresses.. I can add it back if we think the space complexity is worth